/*
 * This file was automatically generated by EvoSuite
 * Fri May 24 09:30:20 GMT 2024
 */

package Controller;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.shaded.org.mockito.Mockito.*;
import static org.evosuite.runtime.EvoAssertions.*;
import Controller.ControladorUtilizador;
import Model.Vintage;
import View.Apresentacao;
import java.util.NoSuchElementException;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.ViolatedAssumptionAnswer;
import org.evosuite.runtime.util.SystemInUtil;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class ControladorUtilizador_ESTest extends ControladorUtilizador_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      SystemInUtil.addInputLine("3");
      ControladorUtilizador controladorUtilizador0 = new ControladorUtilizador();
      Apresentacao apresentacao0 = mock(Apresentacao.class, new ViolatedAssumptionAnswer());
      Vintage vintage0 = mock(Vintage.class, new ViolatedAssumptionAnswer());
      // Undeclared exception!
      try { 
        controladorUtilizador0.interpretador(vintage0, apresentacao0);
        fail("Expecting exception: NoSuchElementException");
      
      } catch(NoSuchElementException e) {
         //
         // No line found
         //
         verifyException("java.util.Scanner", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      SystemInUtil.addInputLine("1");
      ControladorUtilizador controladorUtilizador0 = new ControladorUtilizador();
      Vintage vintage0 = mock(Vintage.class, new ViolatedAssumptionAnswer());
      Apresentacao apresentacao0 = mock(Apresentacao.class, new ViolatedAssumptionAnswer());
      // Undeclared exception!
      try { 
        controladorUtilizador0.interpretador(vintage0, apresentacao0);
        fail("Expecting exception: NoSuchElementException");
      
      } catch(NoSuchElementException e) {
         //
         // No line found
         //
         verifyException("java.util.Scanner", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      SystemInUtil.addInputLine("2");
      Vintage vintage0 = mock(Vintage.class, new ViolatedAssumptionAnswer());
      Apresentacao apresentacao0 = mock(Apresentacao.class, new ViolatedAssumptionAnswer());
      ControladorUtilizador controladorUtilizador0 = new ControladorUtilizador();
      // Undeclared exception!
      try { 
        controladorUtilizador0.interpretador(vintage0, apresentacao0);
        fail("Expecting exception: NoSuchElementException");
      
      } catch(NoSuchElementException e) {
         //
         // No line found
         //
         verifyException("java.util.Scanner", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      SystemInUtil.addInputLine("0");
      Vintage vintage0 = mock(Vintage.class, new ViolatedAssumptionAnswer());
      Apresentacao apresentacao0 = mock(Apresentacao.class, new ViolatedAssumptionAnswer());
      ControladorUtilizador controladorUtilizador0 = new ControladorUtilizador();
      String string0 = controladorUtilizador0.interpretador(vintage0, apresentacao0);
      assertNull(string0);
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      ControladorUtilizador controladorUtilizador0 = new ControladorUtilizador();
      // Undeclared exception!
      try { 
        controladorUtilizador0.interpretador((Vintage) null, (Apresentacao) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("Controller.Input", e);
      }
  }
}
